src/filters/medianFilter.ts

```diff
@@ -0,0 +1,82 @@
+import { xMedian } from 'ml-spectra-processing';
+
+import { Image } from '../Image';
+import checkProcessable from '../utils/checkProcessable';
+import { getBorderInterpolation, BorderType } from '../utils/interpolateBorder';
+
+export interface MedianFilterOptions {
+  /**
+   * Type of border algorithm to interpolate from.
+   *
+   * @default 'reflect101'
+   */
+  borderType: BorderType;
+  /**
+   * Value of border.
+   */
+  borderValue: number;
+  /**
+   * The radius of the cell to extract xMedian from. Must be odd.
+   *
+   * @default 1
+   */
+  cellSize: number;
+}
+/**
+ * Apply a median filter to the image.
+ *
+ * @param image - Image to be filtered.
+ * @param options - MedianFilterOptions
+ * @returns Image after median filter.
+ */
+export function medianFilter(image: Image, options: MedianFilterOptions) {
+  let { cellSize = 1, borderType = 'reflect101', borderValue } = options;
+
+  checkProcessable(image, {
+    bitDepth: [8, 16],
+  });
+
+  if (cellSize < 1) {
+    throw new RangeError('cellSize must be greater than 0');
+  }
+
+  if (cellSize % 2 === 0) {
+    throw new Error('cellSize must be an odd number');
+  }
+
+  // validateChannels(options.channels as number[], image);
+  let interpolateBorder = getBorderInterpolation(borderType, borderValue);
+  let kSize = cellSize;
+  let newImage = Image.createFrom(image);
+  let size = (kSize * 2 + 1) * (kSize * 2 + 1);
+  let kernel;
```

Comment:
Just define it always as new Uint16Array(size). It may always be problematic that the variable may be 2 different kinds.

Reply:
`kernel` seems a strange name. Maybe `cellValues`